sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/ConfigurationController.java

```diff
@@ -11,14 +20,42 @@
 class ConfigurationController {
 
   private static final ConfigurationApi CLIENT = new ConfigurationApi();
+  private final ObjectMapper mapper =
+      new ObjectMapper()
+          .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY)
+          .registerModule(new JavaTimeModule());
 
   /**
    * Get the list of configurations.
    *
-   * @return the list of configurations
+   * @param accept the accept header
+   * @return a response entity with a string representation of the list of configurations
    */
   @GetMapping("/configurations")
-  AiConfigurationList getConfigurations() {
-    return CLIENT.query("default");
+  ResponseEntity<String> getConfigurations(
+      @RequestHeader(value = "accept", required = false) final String accept)
+      throws JsonProcessingException {
+    final var configList = CLIENT.query("default");
+    if ("application/json".equals(accept)) {
+      return ResponseEntity.ok()
+          .contentType(MediaType.APPLICATION_JSON)
+          .body(mapper.writeValueAsString(configList));
+    }
+    return ResponseEntity.ok(buildMessage(configList));
+  }
+
+  /**
+   * Build a message from the configuration list.
+   *
+   * @param configList The configuration list.
+   * @return the message
+   */
+  private String buildMessage(final AiConfigurationList configList) {
+    final var message = new StringBuilder("The following configurations are available: ");
+    for (final var resource : configList.getResources()) {
+      message.append(resource.getName()).append(", ");
+    }
+    message.setCharAt(message.length() - 2, '.');
+    return message.toString();
```

Comment:
(Minor)

Code can be reduced.

```suggestion
    final var items =
        configList.getResources().stream().map(AiConfiguration::getName).collect(joining(", "));
    return ResponseEntity.ok("The following configurations are available: %s.".formatted(items));
```

with
```
import static java.util.stream.Collectors.joining;
```

---

Google Commons library also has some nice convenience API. But I wouldn't advertise this in a sample app:
```
final var items =
  Joiner.on(", ").join(Iterables.transform(configList.getResources(), AiConfiguration::getName));
```